### PR TITLE
Migrate Persistent Internal Dashboard

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot from 2022-04-08 15-11-31
Screenshot from 2022-04-08 15-11-19

### CODE REVIEW

File: packages/shared/dashboards/utilization-card/ConsumerPopover.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      useK8sWatchResource<K8sResourceCommon[]>(k8sResource);
+
+    const [metrics, metricsError, metricsLoading] = isOpen
+      ? // eslint-disable-next-line react-hooks/rules-of-hooks
+        usePrometheusPoll({
+          endpoint: 'api/v1/query' as any,
+          query,
+          namespace,
+        })
+      : [null, null, false];
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/shared/dashboards/utilization-card/ConsumerPopover.tsx]: yeah I was experimenting turns out this isn't the best way to achieve it.

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import { getCephHealthState, getDataResiliencyState } from './utils';
+import { whitelistedHealthChecksRef } from './whitelisted-health-checks';
+import './healthchecks.scss';
+
+const resiliencyProgressQuery =
+  DATA_RESILIENCY_QUERY[StorageDashboardQuery.RESILIENCY_PROGRESS];
+
+export const CephAlerts: React.FC = () => {
+  const [data, loaded, error] = useAlerts();
+  const alerts = data ? filterCephAlerts([]) : [];
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:42]:
```suggestion
  const alerts = data ? filterCephAlerts(data?.data?.result) : [];
```

File: packages/ocs/dashboards/persistent-internal/utilization-card/area-chart.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
+import * as React from 'react';
+import {} from '@odf/shared/charts/PrometheusGraph';
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/utilization-card/area-chart.tsx:2]: ??

File: packages/shared/charts/PrometheusGraph.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+}) => {
+  const [perspective] = useActivePerspective();
+  const queries = _.compact(_.castArray(query));
+  if (!queries.length) {
+    return <>{children}</>;
+  }
+
+  const params = new URLSearchParams();
+  queries.forEach((q, index) => params.set(`query${index}`, q));
+
```
Comment (SanjalKatiyar) [inline packages/shared/charts/PrometheusGraph.tsx:40]:
```
// @ts-ignore (typescript error: Property 'SERVER_FLAGS' does not exist on type 'Window & typeof globalThis') 
const canAccessMonitoring = useFlag(CAN_GET_NS) && !!window.SERVER_FLAGS.prometheusBaseURL;
```
this should work...

**Pull request merged**